internal/photoprism/mediafile.go

```diff
@@ -314,6 +317,86 @@ func (m *MediaFile) EditedName() string {
 	return ""
 }
 
+func (m *MediaFile) ExtractEmbeddedVideo() (string, error) {
+	if m == nil {
+		return "", fmt.Errorf(
+			"mediafile: file is nil - you may have found a" +
+				" bug",
+		)
+	}
+
+	// Abort if the source media file does not exist.
+	if !m.Exists() {
+		return "", fmt.Errorf(
+			"mediafile: %s not found", clean.Log(m.RootRelName()),
+		)
+	} else if m.Empty() {
+		return "", fmt.Errorf(
+			"mediafile: %s is empty", clean.Log(m.RootRelName()),
+		)
+	}
+
+	// get the embedded video field name from the file metadata
+	if metaData := m.MetaData(); metaData.Error == nil && metaData.
+		EmbeddedVideo != "" {
+		path := filepath.Join(
+			Config().SidecarPath(), m.RootRelPath(), "%f",
+		)
+		cmd := exec.Command(
+			Config().ExifToolBin(), fmt.Sprintf("-%s", metaData.EmbeddedVideo),
+			"-b", "-w",
+			path, m.FileName(),
+		)
+
+		var out bytes.Buffer
+		var stderr bytes.Buffer
+		cmd.Stdout = &out
+		cmd.Stderr = &stderr
+		cmd.Env = []string{fmt.Sprintf("HOME=%s", Config().CmdCachePath())}
+		if err := cmd.Run(); err != nil {
+			log.Debugf("Error running exiftool on video file: ", err)
+			if stderr.String() != "" {
+				return "", errors.New(stderr.String())
+			} else {
+				return "", err
+			}
+		}
+
+		// find the extracted video path
+		savedPath := filepath.Join(
+			Config().SidecarPath(),
+			m.RootRelPath(),
+			m.BasePrefix(false),
+		)
+
+		// detect mime type of the extracted video
+		mimeType := fs.MimeType(savedPath)
+
+		if len := len(strings.Split(mimeType, "/")); len <= 1 {
+			log.Debugf(
+				"Error detecting the mime type of sidecar file at %s",
+				savedPath,
+			)
+			return "", nil
+		} else if extension := strings.Split(
+			mimeType, "/",
+		)[len-1]; extension != "" {
+			// rename the extracted video file with the correct extension
+			dir, file := filepath.Split(savedPath)
+			newFileName := fmt.Sprintf("%s.%s", file, extension)
+			newSavedPath := filepath.Join(dir, newFileName)
+			err := os.Rename(savedPath, newSavedPath)
```

Comment:
`Rename` fails when you try to move a file across different filesystems. We added a wrapper in the `fs` package that reverts to Copy & Remove in this case:
https://github.com/photoprism/photoprism/blob/1507525ba4a3ebfc96cb2c65ee9493740bd2cf59/pkg/fs/move.go#L10-L34